api-registration/server/index.js

```diff
@@ -0,0 +1,63 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+
+  /* your code starts here */
+
+  /**
+   * Hash the user's password with `argon2.hash()`
+   * Then, 😉
+   *   Insert the user's "username" and "hashedPassword" into the "users" table.
+   *   Then, 😉
+   *     Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+   *   Catch any errors.
+   * Catch any errors.
+   *
+   * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+   */
+
+  const sql = `
+  INSERT INTO "users" ("username", "hashedPassword")
+  VALUES ($1, $2)
+  RETURNING "username", "userId", "createdAt"
+      `;
+  argon2
+    .hash(password)
+    .then(hashedPassword => {
+      const values = [username, hashedPassword];
+      db.query(sql, values)
+        .then(result => {
+          res.status(201).json(result.rows);
```

Comment:
This will send back an array with one object inside of it, but we only want to send back the object. So we want to do `rows[0]`.
```JAVASCRIPT
res.status(201).json(result.rows[0]);
```